src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -1,18 +1,33 @@
 (ns status-im.contexts.wallet.swap.events
   (:require [re-frame.core :as rf]
             [status-im.constants :as constants]
+            [status-im.contexts.communities.utils :as utils]
             [status-im.contexts.wallet.sheets.network-selection.view :as network-selection]
             [utils.number]))
 
 (rf/reg-event-fx :wallet.swap/start
- (fn [{:keys [_db]}]
+ (fn [{:keys [_]}]
    {:fx [[:dispatch [:open-modal :screen/wallet.swap-select-asset-to-pay]]]}))
 
+(rf/reg-event-fx :wallet.swap/start-with-token
+ (fn [{:keys [db]} [token]]
+   (let [current-address (get-in db [:wallet :current-viewing-account-address])
+         address         (:address (first (utils/sorted-operable-non-watch-only-accounts db)))
+         token-networks  (:networks token)]
+     {:fx [(when-not current-address
+             [:dispatch
+              [:wallet/switch-current-viewing-account address]])
+           [:dispatch
+            [:wallet.swap/select-asset-to-pay
+             {:token   token
+              :network (when (= count token-networks)
```

Comment:
I am not sure about this code, it seems that here we want to check token-networks is empty or not or if its atleast more than 0. How the `count` function is used here does not explicitly make this comparisson. 

I suggest to use `(not (empty? token-networks)` if the intent is to check that there is some networks.

Reply:
Exactly @jo-mut , we are comparing a function (`count`) with `token-networks`, this is going to be false all the time,

> I suggest to use (not (empty? token-networks) if the intent is to check that there is some networks.

`(seq token-networks)` is the recommended use to check if something is not empty.